### PR TITLE
[Fix #448] Delete a publication close readers of this publication

### LINKED ISSUE #448: All reader views for a given publication should be closed before deleting the publication from the bookshelf
In the current implementation, reader views remain open, and will attempt to access deleted resources from the filesystem (EPUB contents).

### PR DESCRIPTION
**Pull request by Scarsniik**
fix #448

### CODE REVIEW

**Comment by danielweck** [comment]
This is great, thank you. Could you please make sure the commits "[fix] locale persistence" (from Pierre) and "Merge branch 'develop' of github.com:edrlab/readium-desktop into HEAD" (from Laurent) do not appear in your PR's commit history?

**Comment by Scarsniik** [comment]
I did it. I don't know why it was like that

File: src/main/redux/sagas/reader.ts [hunk 1]
```diff
@@ -193,58 +198,80 @@ export function* readerCloseRequestWatcher(): SagaIterator {
...
-
-        if (gotoLibrary) {
-            // Show library window
-            const appWindows = winRegistry.getWindows();
-
-            for (const appWin of Object.values(appWindows)) {
-                if (appWin.type !== AppWindowType.Library) {
-                    continue;
-                }
+function* closeReader(reader: any, gotoLibrary: boolean): any {
```
Comment (danielweck) [inline src/main/redux/sagas/reader.ts]: Can this `any` be replaced with a proper TypeScript type?
`Reader` from `src/common/models/reader.ts`?
https://github.com/readium/readium-desktop/blob/7776294d83aae7184eee47ccfa06adde13789d14/src/common/models/reader.ts#L21-L26

Comment (Scarsniik) [reply src/main/redux/sagas/reader.ts]: I deleted the return type, it was foor something I wanted to do but I choosed an other way to do it and didn't delete it

Comment (danielweck) [reply src/main/redux/sagas/reader.ts]: Okay thank. What about `reader: any`?

Comment (Scarsniik) [reply src/main/redux/sagas/reader.ts]: Oh, sorry, I didn't saw it. I fixed it

**Comment by danielweck** [comment]
Hotfix: https://github.com/readium/readium-desktop/commit/926000f851597e4cbe6cd3b51a8748ce585b7790

See code comment: https://github.com/readium/readium-desktop/commit/bc3f1337bf50c857bf7a0a48f57e1c076056a697#r35171002

**Pull request merged**